test/tests/restart/restart_subapp_not_master/two_step_solve_master.i

```diff
@@ -0,0 +1,105 @@
+[Mesh]
+  type = GeneratedMesh
+  dim = 2
+  xmin = -1
+  xmax =  1
+  ymin = -1
+  ymax =  1
+  nx = 2
+  ny = 2
+  elem_type = QUAD9
+[]
+
+[Functions]
+  [./exact_fn]
+    type = ParsedFunction
+    value = t*t*(x*x+y*y)
+  [../]
+
+  [./forcing_fn]
+    type = ParsedFunction
+    value = 2*t*(x*x+y*y)-4*t*t
+  [../]
+[]
+
+[Variables]
+  [./u]
+    family = LAGRANGE
+    order = SECOND
+  [../]
+[]
+
+[ICs]
+  active = ''
+  [./u_var]
+    type = FunctionIC
+    variable = u
+    function = exact_fn
+  [../]
+[]
+
+[Kernels]
+  [./td]
+    type = TimeDerivative
+    variable = u
+  [../]
+
+  [./diff]
+    type = Diffusion
+    variable = u
+  [../]
+
+  [./ffn]
+    type = UserForcingFunction
+    variable = u
+    function = forcing_fn
+  [../]
+[]
+
+[BCs]
+  [./all]
+    type = FunctionDirichletBC
+    variable = u
+    boundary = 'left right top bottom'
+    function = exact_fn
+  [../]
+[]
+
+[Postprocessors]
+  [./average]
+    type = ElementAverageValue
+    variable = u
+    execute_on = 'initial timestep_end'
+  [../]
+[]
+
+[Executioner]
+  type = Transient
+  start_time = 2.0
+  end_time = 4.0
+  dt = 1.0
+[]
+
+[MultiApps]
+  [./full_solve]
+    type = FullSolveMultiApp
+    execute_on = initial
+    positions = '0 0 0'
```

Comment:
Since @permcody added one. This line is not needed after my recent PR in moose. I see you did not put `app_type`, hehe. You missed this one.